src/modals/CreatingStudyModal.js

```diff
@@ -0,0 +1,341 @@
+      } else if (bucketEvents[0] === 'BK_STR') {
+        return 'Creating';
+      }
+    }
+    if (studyByKfId && studyByKfId.bucket) {
+      return 'Exist';
+    }
+  })();
+
+  // Set project creation status based on project related events
+  const projectStatus = (() => {
+    if (!projectEvents || projectEvents.length <= 0) {
+      return 'Pending';
+    } else {
+      if (projectEvents[0] === 'PR_SUC') {
+        return 'Created';
+      } else if (projectEvents[0] === 'PR_ERR') {
+        return 'Failed';
+      } else {
+        return 'Creating';
+      }
+    }
+  })();
+  var newStudies = sessionStorage.getItem('newStudy')
+    ? sessionStorage.getItem('newStudy').split(',')
+    : [];
+  const currentIndex =
+    newStudies.indexOf(newStudies.filter(id => id.includes(studyKfId))[0]) || 0;
+
+  // Set overall study creation status based on bucket status and project status
+  const studyStatus = (() => {
+    if (bucketStatus === 'Failed' || projectStatus === 'Failed') {
+      newStudies[currentIndex] = studyKfId + '_ERR';
+      sessionStorage.setItem('newStudy', newStudies);
+      return 'Failed';
+    }
+    if (
+      (bucketStatus === 'Created' && projectStatus === 'Pending') ||
+      (projectStatus === 'Created' && bucketStatus === 'Pending') ||
+      (bucketStatus === 'Created' && bucketStatus === 'Created')
+    ) {
+      newStudies[currentIndex] = studyKfId + '_SUC';
+      sessionStorage.setItem('newStudy', newStudies);
+      return 'Created';
+    } else {
+      newStudies[currentIndex] = studyKfId + '_STR';
+      sessionStorage.setItem('newStudy', newStudies);
+      return 'Creating';
+    }
+  })();
+
+  if (studyStatus === 'Creating') {
+    eventsStartPulling(1000);
+    studyStartPulling(1000);
+    setTimeout(() => {
+      eventsStopPulling();
+      studyStopPulling();
+    }, 30000);
+  } else {
+    eventsStopPulling();
+    studyStopPulling();
+  }
+
+  const statusComponent = {
+    Creating: <List.Icon name="spinner" loading className="noPadding" />,
+    Created: <List.Icon name="check" className="noPadding" color="green" />,
+    Failed: <List.Icon name="x" className="noPadding" color="red" />,
+    Pending: <List.Icon name="clock outline" className="noPadding text-grey" />,
+    Exist: <List.Icon name="check" className="noPadding" color="green" />,
+  };
+
+  return (
+    <Modal
+      open={displaying}
+      onClose={() => {
+        history.push(history.location.pathname);
+        closeModal(false);
+      }}
+      closeIcon
+      size="small"
+    >
+      <Modal.Header>Study Creation Status</Modal.Header>
+      {error ? (
+        <Message
+          negative
+          icon="warning circle"
+          header="Error"
+          content={error.message}
+        />
+      ) : (
+        <Modal.Content>
+          {studyStatus === 'Creating' && (
+            <Header textAlign="center" as="h2" icon className="noBefore">
+              <Icon name="spinner" loading />
```

Comment:
let's replace this with the [Loader](https://react.semantic-ui.com/elements/loader/#variations-inline-centered) we use in the skeleton views